src/flag_gems/ops/avg_pool2d.py

```diff
@@ -0,0 +1,426 @@
+
+            grad_out_ptr = (
+                grad_output_base_ptr + h_out * out_stride_h + w_out * out_stride_w
+            )
+            grad_out_val = tl.load(grad_out_ptr, mask=out_mask, other=0.0)
+            grad_to_add = grad_out_val / divisor
+            grad_acc += tl.where(out_mask, grad_to_add, 0.0)
+
+    grad_input_store_ptr = (
+        grad_input_block_ptr
+        + h_in_offsets[:, None] * in_stride_h
+        + w_in_offsets[None, :] * in_stride_w
+    )
+    in_write_mask = (h_in_offsets[:, None] < in_h) & (w_in_offsets[None, :] < in_w)
+    tl.store(
+        grad_input_store_ptr,
+        grad_acc.to(grad_input_ptr.type.element_ty),
+        mask=in_write_mask,
+    )
+
+
+class AvgPool2d(torch.autograd.Function):
+    @staticmethod
+    def forward(
+        ctx,
+        input,
+        kernel_size,
+        stride,
+        padding,
+        ceil_mode,
+        count_include_pad,
+        divisor_override,
+    ):
+        logger.debug("GEMS AVG_POOL2D FORWARD")
+        input = input.contiguous()
+
+        if isinstance(kernel_size, int):
+            kernel_h = kernel_w = kernel_size
+        else:
+            kernel_h, kernel_w = kernel_size
+        if stride is None or (isinstance(stride, (list, tuple)) and not stride):
+            stride_h, stride_w = kernel_h, kernel_w
+        elif isinstance(stride, int):
+            stride_h = stride_w = stride
+        else:
+            stride_h, stride_w = stride
+        if isinstance(padding, int):
+            padding_h = padding_w = padding
+        else:
+            padding_h, padding_w = padding
+
+        dilation_h, dilation_w = 1, 1
+
+        in_n, in_c, in_h, in_w = input.shape
+        out_h = pool2d_output_size(
+            in_h, kernel_h, stride_h, padding_h, dilation_h, ceil_mode
+        )
+        out_w = pool2d_output_size(
+            in_w, kernel_w, stride_w, padding_w, dilation_w, ceil_mode
+        )
+
+        output = torch.empty(
+            (in_n, in_c, out_h, out_w), device=input.device, dtype=input.dtype
+        )
+
+        if output.numel() > 0:
+            grid = lambda meta: (
+                in_n * in_c,
+                triton.cdiv(out_h, meta["BLOCK_H"])
+                * triton.cdiv(out_w, meta["BLOCK_W"]),
+            )
+
+            avg_pool2d_forward_kernel[grid](
+                input,
+                output,
+                input.stride(0),
+                input.stride(1),
+                input.stride(2),
+                input.stride(3),
+                in_c,
+                in_h,
+                in_w,
+                out_h,
+                out_w,
+                kernel_h,
+                kernel_w,
+                stride_h,
+                stride_w,
+                padding_h,
+                padding_w,
+                dilation_h,
+                dilation_w,
+                COUNT_INCLUDE_PAD=count_include_pad,
+                divisor_override=divisor_override or 0,
```

Comment:
I advice to check that divisor is not zero if it is not None.
If it is None, just pass it to the jit function. Do not use 0 as a magic number.